### DISCUSSION TITLE
✏️ [ FEAT REQ ] GitHub Wiki or Other Feature Documentation

### DISCUSSION THREAD

**Discussion by Pinaz993** [octoshrimpy/quik#174] [General]
qkSMS and quik both have very sparse documentation. (At least, I was not able to find any documentation other than the README.MD.) In fact, I have noticed two instances where people have wanted a feature, unaware that the feature is already implemented.

While it may seem that an SMS/MMS client does not need much in the way of documentation, if written and organized carefully, unneeded docs can't hurt.

In addition, while it may be daunting to dive into Java/Kotlin and learn how to handle *shudder* Gradle, Markdown is actually quite approachable.

I propose that we, the grateful users of quik, take some of the load off of @octoshrimpy and work to create documentation of the features in the app in such a way that it can be later implemented or linked to in the app itself. 

This issue may be useful as a gathering point for information on features of qkSMS and quik.

**Comment by Pinaz993** [comment]
I'll start with the two features I saw that needed documenting:
- scheduled messages (#69)
- Group text vs. multi-text switch when multiple numbers/contacts are in the recipient field (#170)

**Comment by Pinaz993** [comment]
Many who are not in the know may wonder about RCS support. A paragraph or so should be witten about why this will not happen. (If the situation changes, this can be removed from the documentation.)

**Reply by julien-garrigue** [reply]
Do you mean that Quik will never support RCS?

**Reply by Inhishonor** [reply]
@julien-garrigue At the moment, it is not likely that RCS will be supported. This issue is being tracked in #124. We wanted a section in the wiki mainly to explain the reasons and redirect to the issue where the discussion is taking place.

**Reply by Inhishonor** [reply]
Added in the [FAQ](https://github.com/octoshrimpy/quik/wiki/Frequently-Asked-Questions). Does it look good?

**Comment by Pinaz993** [comment]
A development roadmap/planned features list could not only cut down on questions, but also serve as a sort of issue index of sorts, with links to each feature request issue.

**Comment by octoshrimpy** [comment]
This is fantastic, thank you so much! :purple_heart: 

I'm working on a small roadmap for "next release" over at  https://github.com/users/octoshrimpy/projects/4/views/1

and for those coming in, here's a good place to start for getting friendly with markdown:

https://www.markdownguide.org

**Comment by Pinaz993** [comment]
> https://github.com/users/octoshrimpy/projects/4/views/1

This URL returns a `404`, though assuming you see something different, it really should be a `403`.

**Comment by Pinaz993** [comment]
As far as categorization goes, it looks like you can either make a bunch of wiki pages and allow GitHub to simply list them, or you can use Markdown to make a more attractive and better organized sidebar. For example:
```
# Basics
### Installation
- From GitHub
- Using Obtanium

### Usage
- Writing and sending a Text
- Scheduling a text
```
This will make a sidebar that looks like the following:

> # Basics
> ### Installation
> - From GitHub
> - Using Obtanium
>
> ### Usage
> - Writing and sending a Text
> - Scheduling a text

Thus, we have three levels of categorization to work with. I think we'll only need two.

**Comment by Pinaz993** [comment]
As for what to actually put there, I suggest the following: 
 
> # Basics
> ### Installation
> - From GitHub
> - Using Obtanium
> ### Usage
> Messages
> - Sending & Receiving
> - Attachments
>
> # Advanced
> Messages
> - Pinning Conversations
> - Scheduling Messages
> - Archived Messages
> - Multiple Recipients
>
> Side Menu
> - Blocked Senders
> - Message History Backups
> - Inviting Friends
>
> Settings
> - Appearance
> - General
> # Miscellaneous
> FAQ
>
> Troubleshooting
> - Common
> - Issues
> - Go
> - Here
>
> Development
> - Planned Features
> - Contribution

In theory, every wiki page we write should be one click away with this sidebar.

**Comment by Pinaz993** [comment]
Currently, the wiki is disabled. You can enable it in Settings \> Features

**Comment by octoshrimpy** [comment]
@Pinaz993 project made public-read, and wiki made public-RW

**Comment by Pinaz993** [comment]
Wiki landing page created, with struckthrough instructions to navigate with the side bar, an under construction message, and a thanks section based on the one in `README.md`.

**Comment by octoshrimpy** [comment]
we may want to link to https://dontkillmyapp.com/ since a couple people have ran into that problem before: https://www.reddit.com/r/fossdroid/comments/1cjl17i/comment/ljcbvkl/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

**Reply by Inhishonor** [reply]
Added in the [FAQ](https://github.com/octoshrimpy/quik/wiki/Frequently-Asked-Questions).

**Comment by Pinaz993** [comment]
Installation documentation for GitHub and Obtanium Added. Please check and make sure it's clear and understandable.

**Comment by Pinaz993** [comment]
Sorry I couldn't do anything for a couple of days. Initial Setup page written, with screenshot. Please make sure it is clear and understandable.

**Comment by Pinaz993** [comment]
Basic usage is finally done and linked in the sidebar! As it turns out, writing Markdown with a phone is not nearly as frustrating and annoying as I expected it to be.

**Comment by octoshrimpy** [comment]
@Pinaz993 your work is very much appreciated!!

**Comment by Inhishonor** [comment]
Do you mind if I work on the documentation to? I could add the sections about archiving, scheduling and blocking.

**Comment by Pinaz993** [comment]
I have no problem with it, and I don't think our dear dev will have a
problem either, considering how he reacted to my offer.

**Reply by Inhishonor** [reply]
Great! I will work on it.

**Comment by Inhishonor** [comment]
I added:
- Scheduling Messages
- Archiving Messages
- Blocking Numbers
I also added a link in the Menu to a page for sorting messages based on this pull request. #213  Please double check that it is clear and there are no typos. Thanks!

**Reply by Pinaz993** [reply]
Wunderbar! Thank you very much for your contribution, good sir!

**Comment by octoshrimpy** [comment]
@Inhishonor thank you for all your help! in my book you are a sir/madam and a scholar!

**Comment by Inhishonor** [comment]
Added `Sorting Messages`, `Backup and Restore` and set up the Settings page.  At what point do you think that the wiki is complete enough to add links in the F-Droid description, Readme and within the app? Also, do you think we should add a thank you to @gavine99 in the Thank you section, given the amount of amazing features they have added? Thanks!

**Comment by Inhishonor** [comment]
I added renaming conversations, inviting friends and adding a contributing guidelines. @octoshrimpy Could you take a look at the contributing guidelines especially. I know it is customary to have the project owner create that, so I wanted to make sure you liked it. Also, I still haven't figured out how to get it to build locally, and so if you could possibly write instructions for that, that would be really helpful. Does everything else look good?

**Comment by connect-mahdi** [comment]
Hi all ! What a GREAT project !

I hope you don't mind if I indicate here, as a QUIK user, that there is a very useful backup app for SMS/MMS and much more:
You might take a look at this project for backup purpose [https://github.com/tmo1/sms-ie](https://github.com/tmo1/sms-ie)

**Reply by Inhishonor** [reply]
Assuming everyone else agrees, I will add it as an option. Thanks!

**Comment by connect-mahdi** [comment]
Backup and Restore being part of one whole process, it might be helpful to include details about the restoring part and it's several use cases

**Comment by Inhishonor** [comment]
@octoshrimpy @Pinaz993 I added a bunch of pages to the Wiki. Settings, Voice Actions, FAQ. I also filled out the development section, and added a section in the installation page, about the restricted settings annoyance. Does it look good? What else can be added to the wiki? Thanks!

**Comment by Pinaz993** [comment]
It looks very good. Thank you for all your work on this. We'd have very
little Wiki without your efforts.

**Reply by Inhishonor** [reply]
No problem. Also, thank you for spearheading this and setting up a template. It made it really easy to add sections.
